lib/zclient.c

```diff
@@ -749,10 +749,24 @@ int zapi_route_encode(uint8_t cmd, struct stream *s, struct zapi_route *api)
 	stream_reset(s);
 	zclient_create_header(s, cmd, api->vrf_id);
 
+	if (api->type >= ZEBRA_ROUTE_MAX) {
```

Comment:
Yeah I intentionally removed these checks after thinking about them.  Why are we adding them back in?

Reply:
I think this one in particular is a new sanity check (please correct me if I'm wrong). We're already checking the same thing on `zapi_route_decode()`, so I think it makes sense to do this check here as well. Since route types are not defined using an enum this should be fine. What do you think?

Reply:
I don't see a problem with this and I think it makes sense to have a check that occurs in the client process as well as the check in the zebra process. As an aside, if we do that, keeping what we do and don't check for exactly the same between Zebra and `zclient.c` would be ideal.